test/parallel/test-stream2-compatibility.js

```diff
@@ -44,7 +44,7 @@ class TestReader extends R {
 }
 
 const reader = new TestReader();
-setImmediate(function() {
+process.nextTick(function() {
```

Comment:
This test kind of contradicts the other test, i.e. it expects the stream not live lock when pushing empty chunks.

Reply:
Can you please articulate this a bit more? Is this change needed to make the test pass?

Reply:
Yes, otherwise it live locks. Basically it gets into an infinite loop (with nextTick) between. Because it pushed an empty chunk, which invokes a read call which pushes an empty chunk etc... 

By changing it to nextTick instead of setImmediate it's possible to get between that loop and end it.

Reply:
I think we should avoid that live lock in code.

Reply:
> I think we should avoid that live lock in code.

What does that mean in practice? Should we add a guard? Or is the current state of the PR ok in that regard?

Reply:
I think @mcollina would like to add a guard in the streams implementation that this can not cause an infinite loop / that this test does not need to change.